poh/src/poh_recorder.rs

```diff
@@ -977,34 +985,21 @@ impl PohRecorder {
         self.clear_bank();
     }
 
-    pub fn working_bank_block_producer_time_nanos(&self) -> u64 {
-        self.working_bank()
-            .unwrap()
+    pub fn tick_alpenglow(&mut self, slot_max_tick_height: u64, mut footer: BlockFooterV1) {
+        // Update the footer's timestamp
+        let working_bank = self.working_bank.as_ref().unwrap();
+        let block_producer_time_nanos = working_bank
             .start
             .duration_since(UNIX_EPOCH)
             .expect("Misconfigured system clock; couldn't measure block producer time.")
-            .as_nanos() as u64
-    }
-
-    fn produce_block_footer(&self, working_bank: &WorkingBank) -> VersionedBlockMarker {
-        if !working_bank.bank.is_frozen() {
-            let slot = working_bank.bank.slot();
-            error!("slot = {slot} creating a block footer with a non-frozen bank! ");
-        }
+            .as_nanos() as u64;
+        footer.block_producer_time_nanos = block_producer_time_nanos;
 
-        let footer = BlockFooterV1 {
-            bank_hash: working_bank.bank.hash(),
-            block_producer_time_nanos: self.working_bank_block_producer_time_nanos(),
-            block_user_agent: format!("agave/{}", version!()).into_bytes(),
-        };
-
-        let footer = VersionedBlockFooter::Current(footer);
-        let footer = BlockMarkerV1::BlockFooter(footer);
-
-        VersionedBlockMarker::Current(footer)
-    }
+        BlockComponentProcessor::update_bank_with_footer(
```

Comment:
dunno if it's possible to organize it this way but it would be really nice if we could just call `on_marker` (or if you're planning to put processing in a separate fn something like `process_marker`) here, so that there's no possibility of divergence with the non-leader code

Reply:
yeah makes sense, atm header doesn't affect any state change so I guess not really a point in processing it on the leader side.

I suppose if we wanted we could use the same idea of creating the header in block creation loop and processing it after creating the new bank and then letting broadcast fill in the parent block id